### PR TITLE
feat: custom overlay scrollbar, fixes #1045

### LINKED ISSUE #1045: [Bug] Hover highlight in entity explorer should be edge-to-edge
## Steps
1. Hover over an item in the entity explorer in an application
2. The hover is not edge to edge

### PR DESCRIPTION
**Pull request by sumanthyedoti**
fixes #1045 

### Type of changes
- Feature (custom scrollbar without gutter)
- Bug fix

### How Has This Been Tested?
- Visually tested

### CODE REVIEW

File: app/client/src/pages/Editor/Explorer/index.tsx [hunk 1]
```diff
@@ -25,7 +25,7 @@ import { getPlugins } from "selectors/entitiesSelector";
 
 const Wrapper = styled.div`
   height: 100%;
-  overflow-y: scroll;
+  overflow-y: auto;
```
Comment (riodeuno) [inline app/client/src/pages/Editor/Explorer/index.tsx:28]: This was intentionally left as `scroll`, as without it, when the contents of the explorer have a height greater than the viewport height, the entity explorer reduces in width to make way for the scrollbar. This results in a "jump" noticed in the explorer.

entityexplorerjump

We need to figure out a solution which handles both the "jump" as well has "edge-to-edge" highlight.

Comment (sumanthyedoti) [reply app/client/src/pages/Editor/Explorer/index.tsx:28]: Sure. Thanks for mentioning this. Will try to resolve the jump.

Comment (sumanthyedoti) [reply app/client/src/pages/Editor/Explorer/index.tsx:29]: Removed the scrollbar completely and made a reusable custom scrollbar, not to add another dependency like [simplebar](https://github.com/Grsmto/simplebar).
Inspiration for this is the scrollbar in Webflow's editor.

Comment (sumanthyedoti) [reply app/client/src/pages/Editor/Explorer/index.tsx:29]: Removed the browser scrollbar completely and made a reusable custom overlay scrollbar, without making the gutter.

Comment (riodeuno) [reply app/client/src/pages/Editor/Explorer/index.tsx:29]: @sumanthyedoti  This is a nice approach!

**Review by riodeuno — APPROVED** [review]

**Comment by riodeuno** [comment]
Looks good to me. 
@sumanthyedoti Could you resolve the merge conflict, so that we can merge this in?

**Comment by sumanthyedoti** [comment]
@riodeuno  Will do

**Comment by jsartisan** [comment]
@sumanthyedoti Hey, Your branch had some merge conflicts. A lot of code has been done since November. So couldn't fix the merge conflicts with the web editor. Created a new PR with the same changes - https://github.com/appsmithorg/appsmith/pull/2478

Thanks a lot for the PR. The fix is working as expected :D

**Pull request closed without merging**